src/effects/Stabilizer.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file
+ * @brief Header file for Stabilizer effect class
+ * @author Jonathan Thomas <jonathan@openshot.org>
+ *
+ * @ref License
+ */
+
+/* LICENSE
+ *
+ * Copyright (c) 2008-2019 OpenShot Studios, LLC
+ * <http://www.openshotstudios.com/>. This file is part of
+ * OpenShot Library (libopenshot), an open-source project dedicated to
+ * delivering high quality video editing and animation solutions to the
+ * world. For more information visit <http://www.openshot.org/>.
+ *
+ * OpenShot Library (libopenshot) is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * OpenShot Library (libopenshot) is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with OpenShot Library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPENSHOT_STABILIZER_EFFECT_H
+#define OPENSHOT_STABILIZER_EFFECT_H
+
+#include "../EffectBase.h"
+
+#include <google/protobuf/util/time_util.h>
+
+#include <cmath>
+#include <stdio.h>
+#include <memory>
+#include "../Color.h"
+#include "../Json.h"
+#include "../KeyFrame.h"
+#include "stabilizedata.pb.h"
+
+using namespace std;
+using google::protobuf::util::TimeUtil;
+
+// Store the relative transformation parameters between consecutive frames
+struct EffectTransformParam
+{
+    EffectTransformParam() {}
+    EffectTransformParam(double _dx, double _dy, double _da) {
+        dx = _dx;
+        dy = _dy;
+        da = _da;
+    }
+
+    double dx;
+    double dy;
+    double da; // angle
+};
+
+// Stores the global camera trajectory for one frame
+struct EffectCamTrajectory
+{
+    EffectCamTrajectory() {}
+    EffectCamTrajectory(double _x, double _y, double _a) {
+        x = _x;
+        y = _y;
+        a = _a;
+    }
+
+    double x;
+    double y;
+    double a; // angle
+};
+
+
+namespace openshot
+{
+	
+	/**
+	 * @brief This class stabilizes video clip to remove undesired shaking and jitter.
+	 *
+	 * Adding stabilization is useful to increase video quality overall, since it removes 
+	 * from subtle to harsh unexpected camera movements.
+	 */
+	class Stabilizer : public EffectBase
+	{
+	private:
+		/// Init effect settings
+		void init_effect_details();
+		std::string protobuf_data_path;
+		Keyframe zoom;
+
+	public:
+	std::string teste;
+        std::map <size_t,EffectCamTrajectory> trajectoryData; // Save camera trajectory data
+    	std::map <size_t,EffectTransformParam> transformationData; // Save transormation data 
```

Comment:
You've got a bunch of mixed-spaces-and-tabs damage here. The existing code in this file mostly uses tab indentation currently, personally I prefer four spaces (2 spaces for things like `CMakeLists.txt`, `.js` and `.cs` code, etc.) but it's best to follow the surrounding format when editing existing source.

Since this is a new file, I'm guessing it started as a copy-and-update of one of the other effects headers? In which case, I'm fine with either reformatting the entire thing with space indents, or continuing to use tabs everywhere. But whichever way you go, just pick a lane. :wink:

Reply:
The new `struct`s above are all consistently space-indented. Because they are consistent already, it's fine to leave those as they are, even if the rest of the file continues to be tab-indented. As long as each block is consistent within itself, I don't actually have any objections to mixing within the same file. (Others may disagree, and take a more hard-line approach.)